src/OpenTelemetry.Exporter.InMemory/InMemoryExporterMetricsExtensions.cs

```diff
@@ -36,6 +36,10 @@ public static class InMemoryExporterMetricsExtensions
         /// <param name="builder"><see cref="MeterProviderBuilder"/> builder to use.</param>
         /// <param name="exportedItems">Collection which will be populated with the exported MetricItem.</param>
         /// <returns>The instance of <see cref="MeterProviderBuilder"/> to chain the calls.</returns>
+        /// <remarks>Note that when working with <see cref="Metric"/> these objects are not deep-cloned,
+        /// and repeated flushes may introduce duplicates.
+        /// If you need deep-cloned it is recommended to use <see cref="ExportableMetricCopy"/>.
+        /// </remarks>
         public static MeterProviderBuilder AddInMemoryExporter(this MeterProviderBuilder builder, ICollection<Metric> exportedItems)
```

Comment:
Can this be removed?

Reply:
I agree offering both extension methods can cause confusion.

I can't remove it entirely because the unit tests for Prometheus and OTLP are using this. 
`PrometheusSerializer.WriteMetric()` and `OtlpCollector.ExportMetricsServiceRequest.AddMetrics()` both need the actual `Metric` object.
These tests could possibly be re-written, but I think that should be addressed in a separate PR.

Regarding the PublicAPI, yes! 
This hasn't shipped yet and we can at least mark these as `internal`.

However, _should_ we remove this from the public api?
I don't know this well enough to say if there are legitimate use cases for a public user to have access to the `Metric`.

Reply:
Thinking out loud,
We can remove the extension method and users can still manually create the `InMemoryExporter<Metric>`

Reply:
I would think users should not be able to create an instance of `InMemoryExporter<Metric>` at all.

Reply:
> Thinking out loud, We can remove the extension method and users can still manually create the `InMemoryExporter<Metric>`

This sounds like a bug we should fix.

Reply:
I made a change that allows `InMemoryExporter` to be constructed with a `Func<Batch<T>, ExportResult>` which fully replaces the `Export()` method.
This unblocks Prometheus and OTLP tests which were previously dependent on `InMemoryExporter<Metric>` and allows us to remove those extension methods.